holobot/sdk/utils/dataclass_utils.py

```diff
@@ -41,38 +41,23 @@ def __get_argument_info(
     default_value: Any,
     default_factory: Any
 ) -> ArgumentInfo:
-    if isinstance(object_type, UnionType):
-        origin = UnionType
-    elif (origin := getattr(object_type, "__origin__", None)) is None:
-        return ArgumentInfo(name, object_type, None, False, default_value, default_factory)
-
-    allows_none = False
-    if origin in (Union, UnionType):
-        args = get_args(object_type)
-        if len(args) != 2 or NoneType not in args:
-            raise ValueError((
-                "Expected an optional type (NoneType or other),"
-                f" but '{name}' in '{object_type}' is diferent ({args})."
-            ))
-
-        allows_none = True
-        object_type = first(cast(tuple[type, ...], args), lambda i: i and i is not NoneType)
-        origin = getattr(object_type, "__origin__", None)
-
-    if origin is None:
-        return ArgumentInfo(name, object_type, None, allows_none, default_value, default_factory)
-
-    if origin == tuple:
-        args = get_args(object_type)
-        if len(args) != 2 or args[-1] != Ellipsis:
-            raise ValueError((
-                "Expected a tuple with two arguments, the second being an ellipsis,"
-                f" but got {args} instead."
-            ))
-        return ArgumentInfo(name, args[0], tuple, allows_none, default_value, default_factory)
-
-    if origin == list:
-        args = get_args(object_type)
-        return ArgumentInfo(name, args[0], list, allows_none, default_value, default_factory)
-
-    raise ValueError(f"Expected a tuple or an optional, tuple or list type, but got '{object_type}'.")
+    match origin := get_origin(object_type):
```

Comment:
Are you sure these changes are equivalent?
I'm already at a loss as to what happened to the original first condition
```py
if isinstance(object_type, UnionType):
    origin = UnionType
```

If I remember correctly, this code is tricky with Union and UnionType because `Optional[T]` and `T | None` are constructed differently and you can't check the origin for UnionType, hence `isinstance()`. Reflection is never simple so if you're in doubt, better leave this behavior as it is.